atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/LockWatchValueCacheImpl.java

```diff
@@ -113,53 +119,84 @@ public TransactionScopedCache createTransactionScopedCache(long startTs) {
      */
     private void updateStores(TransactionsLockWatchUpdate updateForTransactions) {
         Multimap<Sequence, StartTimestamp> reversedMap = createSequenceTimestampMultimap(updateForTransactions);
-        updateForTransactions.events().forEach(event -> {
+
+        // There is an edge case where transactions on the current version will not have a snapshot stored.
```

Comment:
what is the edge case

Reply:
Updated comment - basically, if there are no events, we don't store a snapshot (that's without the code below, though; made this much clearer).